src/test/MethodClauseTest.java

```diff
@@ -0,0 +1,103 @@
+package test;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.tools.CustomAssertions;
+import global.tools.Logger;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MethodClauseTest extends BaseTest {
+
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("The sum of two numbers is: "),
+                new IntegerLiteral("sum"),
+                new NewLine(),
+                new StringLiteral("The area is "),
+                new DoubleLiteral("area")
+        };
+    }
+
+    public void runMain() {
+        MethodClause.main(new String[0]);
+    }
+
+    static Stream<String> printInputProvider() {
+        return Stream.of(
+                "Hello",
+                "Goodbye",
+                "Dance till you can no longer insurrectionary"
+        );
+    }
+
+    static Stream<Arguments> addInputProvider() {
+        return Stream.of(
+                Arguments.of(3, 6, 9),
+                Arguments.of(11, 14, 25)
+        );
+    }
+
+    static Stream<Arguments> areaCalcInputProvider() {
+        return Stream.of(
+                Arguments.of(4.5, 3.9, 17.55),
+                Arguments.of(3, 5, 15.0)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("printInputProvider")
+    void correctPrintMethod(String s) throws Throwable {
+        MethodTest m = new MethodTest(MethodClause.class, "print", new Object[]{s}, new Class[]{String.class},
+                new Clause[]{new StringLiteral(s)});
+        m.invokeIfMethodExists();
+    }
+
+    @ParameterizedTest
+    @MethodSource("addInputProvider")
+    void correctAddMethod(int x, int y, int z) throws Throwable {
+        MethodTest m = new MethodTest(MethodClause.class, "add", new Object[]{x, y},
+                new Class[]{int.class, int.class});
+        Object output = m.invokeIfMethodExists();
+        CustomAssertions._assertEquals(z, output, "Your add method does not correctly add two numbers.");
```

Comment:
Does the user see which method we are referring to? If not maybe we should be more specific in the message.

Reply:
I know this is not a real question but when we will ask people to use this as an example of writing method tests. It's good to have it show exactly what we expect in the junits.

Reply:
The user should know what method we are referring to because general if the method is called `add(int x, int y)` we will say that `Your add method does not correctly add two numbers.` .